docs/web/docs/explanations/architect_api.md

```diff
@@ -0,0 +1,108 @@
+- Over any run, the handler should poll with `PACKET_TYPE_REQUEST_STATUS` and update local `Agent` statuses on disconnects from `PACKET_TYPE_RETURN_STATUS`. This also acts as a heartbeat from the Python core to the router. The handler should also take `PACKET_TYPE_ERROR` and log the contents if this ever occurs.
+
+
+### Router Responsibilities
+
+The primary responsiblity of the router is to take incoming packets from client connections and direct them to the core Mephisto `ClientIOHandler` and to do the reverse as well. All packets will have a core `agent_id` field denoting either the sender or receiver of the packet, depending on the packet type. The only exception is the `PACKET_TYPE_ALIVE`, which is directed to the router and allows for any registration of an incoming connection.
+
+Secondarily, the router is responsible for converting RESTful `POST` requests from `mephisto-task` into socket messages, and relaying the response as a standard `POST` response. This behavior is _only_ for the `PACKET_TYPE_REGISTER_WORKER`, and `PACKET_TYPE_SUBMIT` packets, and both of them will be serviced by `PACKET_TYPE_AGENT_DETAILS` responses. For these it should be listening to `POST` requests at `/register_worker`, `submit_onboarding`, and `/submit_task`. `POST` requests to `/log_error` should result in forwarding a `PACKET_TYPE_ERROR`.
+
+Third, the router is responsible for maintaining track of agent status, and acting as a cache for this information after disconnects. This allows for a worker to return to a task and have updated information about what has transpired, even when the main Mephisto server has cleaned up the related `TaskRunner` and live `Agent`.
+
+Fourth, the router is responsible for serving the static `task_config.json` file, which allows the frontend to load certain details about the full task before going through any registration handshakes.
+
+### `mephisto-task` responsibilities.
+
+The `useMephistoTask` hook is responsible for allowing a worker to connect to a task and submit the relevant data. For this, it only needs to make `POST` requests related to the `PACKET_TYPE_SUBMIT` and `PACKET_TYPE_REGISTER_WORKER` events. The former should be triggered on `handleSubmit`, while the latter should trigger immediately on load.
+
+The `useMephistoLiveTask` hook is responsible for the rest of the packets. Data packets should be sent via `sendData` and handled with the `onLiveDataReceived` callback. Providing a callback to `sendData` will call that with any response data that sets the `source_request_id` to the initial send's `message_id`. Local connection state is managed by the `socket_handler.jsx` class, and surfaced through the hook variables for UIs to use. 
```

Comment:
[Request for clarification]

> Providing a callback to `sendData` will call that with any response data that sets the `source_request_id` to the initial send's `message_id`.

I didn't understand this fully. Why would I want to do this? The reader is introduced to `source_request_id` and `message_id` for the first time here, so some more context could be helpful.

Reply:
Yeah this is somewhat vague, but I'm not sure how in-depth I wanted to get here. 

The main thought is that, upon sending a request to the backend, we (`mephisto-task`) attach a `message_id` to the outgoing message. Then the callback handler (yet-to-be-built) should send any resultant messages with a `source_request_id` set (or perhaps `context_id` may be more appropriate), such that `mephisto-task` can later call that specific callback. If no such callback is provided, then only the default `onLiveDataReceived` callback is used. Happy to flesh this out more though.

Reply:
Curious to learn more about a concrete use case where this would be used. I think I may understand what you're getting at but there are still a few gaps in my understanding. The general idea seems to be to link outgoing requests to incoming responses and maintaining a sort of "correlation id" right?